pkg/provider/gitlab.go

```diff
@@ -78,6 +101,15 @@ func (repo *GitLabRepository) Init(config map[string]string) error {
 }
 
 func (repo *GitLabRepository) GetInfo() (*provider.RepositoryInfo, error) {
+	if repo.projectID == os.Getenv("CI_PROJECT_ID") {
+		return &provider.RepositoryInfo{
+			Owner:         os.Getenv("CI_PROJECT_NAMESPACE"),
+			Repo:          os.Getenv("CI_PROJECT_NAME"),
+			DefaultBranch: os.Getenv("CI_DEFAULT_BRANCH"),
+			Private:       os.Getenv("CI_PROJECT_VISIBILITY") != "public",
+		}, nil
+	}
+
```

Comment:
When the `CI_PROJECT_ID` is the same as the configured one we should always use the CI vars instead of calling the API. Only for pipelines which are configured to build another project retrieving the repository info from the API is necessary (when using a job token only public projects can be requested)